src/tests/unit/components/search-input/SearchInput.spec.jsx

```diff
@@ -0,0 +1,61 @@
+import { render, screen, fireEvent } from '@testing-library/react'
+import { vi } from 'vitest'
+import SearchInput from '~/components/search-input/SearchInput'
+import '@testing-library/jest-dom'
+
+describe('SearchInput component', () => {
+  const mockSetSearch = vi.fn()
+
+  afterEach(() => {
+    vi.clearAllMocks()
+  })
+
+  it('should call setSearch when search icon is clicked', () => {
+    render(<SearchInput search='' setSearch={mockSetSearch} />)
+
+    const searchIcon = screen.getByTestId('search-icon')
+    fireEvent.click(searchIcon)
+
+    expect(mockSetSearch).toHaveBeenCalled()
+  })
+
+  it('should call setSearch with an empty string when delete icon is clicked', () => {
+    render(<SearchInput search='test' setSearch={mockSetSearch} />)
+
+    const deleteIcon = screen.getByTestId('delete-icon')
+    fireEvent.click(deleteIcon)
+
+    expect(mockSetSearch).toHaveBeenCalledWith('')
+  })
+
+  it('should call setSearch when enter is pressed', () => {
+    render(<SearchInput search='' setSearch={mockSetSearch} />)
+
+    const input = screen.getByRole('textbox')
+    fireEvent.change(input, { target: { value: 'new value' } })
+    fireEvent.keyPress(input, { key: 'Enter', code: 13, charCode: 13 })
```

Comment:
try to use userEvent type and https://testing-library.com/docs/user-event/keyboard/